utilities/roslz4/src/lz4s.c

```diff
@@ -583,7 +583,10 @@ int roslz4_buffToBuffCompress(char *input, unsigned int input_size,
 
   int ret;
   ret = roslz4_compressStart(&stream, block_size_id);
-  if (ret != ROSLZ4_OK) { return ret; }
+  if (ret != ROSLZ4_OK) {
+    roslz4_compressEnd(&stream);
```

Comment:
It seems kind of weird to call `compressEnd` when `compressStart` is the one that is failing.  I think it might make more sense to fix `roslz4_compressStart` so that if the call to `streamResizeBuffer` fails, it calls `streamStateFree` and then returns the error.  What do you think?

Reply:
Looking at the current implementation of these functions the patch looks good to me (`compressStart` is doing two things which can individually fail). Maybe you can clarify what you think can go wrong with this change?

Reply:
This change will work fine, but I'm suggesting that the contract with `roslz4_compressStart` should be that it either succeeds with some memory allocated, or should fail with no visible side-effects.  That would mean that if something failed internally, it would clean up after itself and not depend on the caller to cleanup for it.  This would also make it easier for any future user of the API to use it, since they wouldn't have to remember to cleanup on failure.

Reply:
I will consider to merge the current patch as-is. Please feel free to refactor the current API / implementation in a separate PR.

Reply:
Sure, that sounds fine.